aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -59,16 +60,22 @@ SparseCsrTensorImpl::SparseCsrTensorImpl(
 void SparseCsrTensorImpl::resize_and_clear_(
     const int64_t nnz_size,
     IntArrayRef size) {
-  // call crow_indices().options() here since the struct contructor calls the
-  // tensor constructor with args for device specific init.
-  auto empty_crow_indices = at::empty(size[0] + 1, crow_indices().options());
-  auto empty_col_indices = at::empty(nnz_size, col_indices().options());
-  auto empty_values = at::empty(nnz_size, values().options());
-
-  crow_indices_ = empty_crow_indices;
-  col_indices_ = empty_col_indices;
-  values_ = empty_values;
+  at::native::resize_output(crow_indices_, size[0] + 1);
+  at::native::resize_output(col_indices_, nnz_size);
+  at::native::resize_output(values_, nnz_size);
   sizes_and_strides_.set_sizes(size);
+
+  TORCH_CHECK(
+      (crow_indices_.numel() - 1) == size[0],
+      "crow_indices.numel() must be size(0) + 1, but got: ",
+      crow_indices_.numel());
+
+  TORCH_CHECK(
+      col_indices_.size(0) == values_.size(0),
+      "col_indices and values must have equal sizes, but got col_indices.size(0): ",
+      col_indices_.size(0),
+      ", values.size(0): ",
+      values_.size(0));
```

Comment:
Why these checks are necessary? Shouldn't we trust `resize_output`?

Reply:
:( Sorry, I don't know what I was thinking you are right is redundant. I will remove it. Taking about invariants, I wonder if there should be an assert for len(size) == 2. I tested it before and it accepts len(size) > 2.

Reply:
Yes, `len(size) == 2` should be forced. But not necessarily in this function nor in this PR.
Could create an issue about `len(size) != 2` non-failure case?

Reply:
Sure!